### PR TITLE
Chapters 19 to 22

### PR DESCRIPTION
**Pull request by SanGeeky**
- try, catch, and rescue
- Typespecs and behaviours
- Debugging
- Erlang libraries

### CODE REVIEW

File: basics/19_typespecs_behaviours.ex [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  @spec current_age(year) :: integer
+  def current_age(year_of_birth), do: IO.puts("YEAR #{year_of_birth}")
+end
+
+
+Person.current_age(2002)
+
+# Custom Types
+defmodule LousyCalculator do
```
Comment (juanhurtado10) [inline basics/19_typespecs_behaviours.ex:18]: @SanGeeky Could you please explain how a behavior works?

File: basics/15_protocols.ex [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+IO.puts(Utility.type("foo"))
+IO.puts(Utility.type(1233))
+
+###################################
+# Example
+###################################
+
+defprotocol Size do
```
Comment (juanhurtado10) [inline basics/15_protocols.ex:25]: @SanGeeky Could you please explain how a protocol works?

File: basics/19_typespecs_behaviours.ex (hunk 1 above)
Comment (SanGeeky) [reply basics/19_typespecs_behaviours.ex:18]: Is an implementation of functions. that ensures the implementation of functions in a module. The behaviour can be used by different modules.

The `@impl` keyword ensures us that the module is implementing all the functions defined in the behaviour, also that the functions contain the correct number of arguments defined in the callback together with their specs.

File: basics/15_protocols.ex (hunk 2 above)
Comment (SanGeeky) [reply basics/15_protocols.ex:25]: Protocols os a kind of polymorphism, and help us to define the action over the data type. Is necessary to define the function that will have the protocol and their implementations for the different data types, also to define a default implementation we can use `Any` to return a default value. The protocols can be used in the same way as modules

**Pull request merged**